pkg/controller/conditions.go

```diff
@@ -172,48 +178,45 @@ func (d *VirtualMachineInstanceConditionManager) AddPodCondition(vmi *v1.Virtual
 	}
 }
 
-func (d *VirtualMachineInstanceConditionManager) PodHasCondition(pod *k8sv1.Pod, conditionType k8sv1.PodConditionType, status k8sv1.ConditionStatus) bool {
+func (d *VirtualMachineInstanceConditionManager) GetPodCondition(pod *k8sv1.Pod, conditionType k8sv1.PodConditionType) *k8sv1.PodCondition {
 	for _, cond := range pod.Status.Conditions {
 		if cond.Type == conditionType {
-			if cond.Status == status {
-				return true
-			} else {
-				return false
-			}
+			return &cond
 		}
 	}
-	return false
+	return nil
 }
 
-func (d *VirtualMachineInstanceConditionManager) GetPodConditionWithStatus(pod *k8sv1.Pod, conditionType k8sv1.PodConditionType, status k8sv1.ConditionStatus) *k8sv1.PodCondition {
-	for _, cond := range pod.Status.Conditions {
-		if cond.Type == conditionType {
-			if cond.Status == status {
-				return &cond
-			} else {
-				return nil
+func (d *VirtualMachineInstanceConditionManager) ConditionsEqual(vmi1, vmi2 *v1.VirtualMachineInstance) bool {
+	if len(vmi1.Status.Conditions) != len(vmi2.Status.Conditions) {
+		return false
+	}
+
+	for _, cond1 := range vmi1.Status.Conditions {
+		found := false
+
+		for _, cond2 := range vmi2.Status.Conditions {
+			if reflect.DeepEqual(cond1, cond2) {
```

Comment:
Did we have a DeepEqual before? This would probably also consider message and timestamp. Something which we normally don't want to avoid hot looping. In some cases we go deliberately as far and prefering an outdated message for a true condition.

Reply:
Yes we've had the DeepEqual comparison before too (see [current implementation](https://github.com/kubevirt/kubevirt/blob/7ecccaa93607a2c103164653476663ad648ca991/pkg/virt-controller/watch/vmi.go#L344)), but I can change it to consider only the `.Type` and `.Value`.

Reply:
Fixed in 6a1335b

Reply:
I think I have to check that in more depth when I have a few calm minutes. I don't want to give false advise.

Reply:
No prob. Keep calm ;-)

Reply:
@rmohr Hi, did you get a chance to consider this?